File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsJava/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicsjava/data/converters/AddressConverter.java
package pl.udu.uwr.pum.offlinecachingbasicsjava.data.converters;

import androidx.room.TypeConverter;

import com.google.gson.Gson;

import pl.udu.uwr.pum.offlinecachingbasicsjava.data.response.Address;

public class AddressConverter {

    @TypeConverter
    public static String fromAddress(Address address){
        Gson gson = new Gson();
        return gson.toJson(address);
    }

    @TypeConverter
    public static Address toAddress(String address){
        Gson gson = new Gson();
        return gson.fromJson(address, Address.class);
    }
}

File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsKotlin/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicskotlin/data/converters/AddressConverter.kt
package pl.udu.uwr.pum.offlinecachingbasicskotlin.data.converters

import androidx.room.TypeConverter
import pl.udu.uwr.pum.offlinecachingbasicskotlin.data.reponse.Address

object AddressConverter {

    @TypeConverter
    @JvmStatic
    fun fromAddress (address: Address): String{
        return address.toString()
    }

    @TypeConverter
    @JvmStatic
    fun toAddress(address: String): Address{
        val city = address.substringAfter("city=").substringBefore(",")
        val country = address.substringAfter("country=").substringBefore(",")
        val state = address.substringAfter("state=").substringBefore(",")
        val street_address = address.substringAfter("street_address=").substringBefore(",")
        val street_name = address.substringAfter("street_name=").substringBefore(",")
        val zip_code = address.substringAfter("zip_code=").substringBefore(")")

        return Address(city, country, state, street_address, street_name, zip_code)
    }
}